monkey/monkey_island/cc/services/authentication_service/authentication_facade.py

```diff
@@ -97,12 +104,17 @@ def generate_refresh_token(self, user: User) -> Token:
         """
         return self._token_generator.generate_token(user.fs_uniquifier)
 
-    def revoke_all_tokens_for_all_users(self):
-        """
-        Revokes all tokens for all users
-        """
-        for user in User.objects:
-            self.revoke_all_tokens_for_user(user)
+    def mark_otp_as_used(self, otp: OTP):
+        self._otp_repository.update_otp(otp, {"used": True})
+
+    def otp_is_valid(self, otp: OTP) -> bool:
+        otp_is_used = self._otp_repository.otp_is_used(otp)
+        expiration_time = self._otp_repository.get_expiration(otp)
+
+        if otp_is_used or expiration_time < time.monotonic():
+            return False
+
+        return True
```

Comment:
We should abstract these two methods behind an `otp_login()` method. That will keep as much logic as possible in the `AuthenticationFacade` and out of the endpoints.

Reply:
In addition, this needs to be locked to avoid the OTP being used multiple times in the event of a race condition.